### ISSUE TITLE
unused-argument in .pyi stub file

### ISSUE THREAD

**Issue by davetapley** [pylint-dev/pylint#9417]
### Bug description

```python
class TestClient:
    def simulate_get(self, path: str) -> Result:
        ...
```

### Command used

```shell
pylint typings/falcon/testing/client.pyi
```

### Pylint output

```python
************* Module client
typings/falcon/testing/client.pyi:13:27: W0613: Unused argument 'path' (unused-argument)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.6 (main, Nov  2 2023, 02:14:33) [GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04.6 LTS

**Comment by jacobtylerwalls** [comment]
Thanks for the report. There are a lot of messages that don't make sense with `.pyi` stubs. No one has compiled a list yet. See #9096 for more discussion. Happy to keep this open as an issue for now, since it's one that will come up a lot.
